lib/northbound.c

```diff
@@ -1056,6 +1056,7 @@ int nb_candidate_apply(struct nb_context *context, struct nb_config *candidate,
 			  "%s: failed to record transaction", __func__);
 
 	nb_transaction_free(transaction);
+	nb_config_diff_del_changes(&changes);
```

Comment:
No. Don't put it here. This forces deletion of the changes with every call to nb_candidate_apply(), which might not be required from other call flows of this function. Let the caller explicitly call it later.